spec/function.dd

```diff
@@ -493,50 +503,69 @@ $(H2 $(LNAME2 pure-functions, Pure Functions))
         All mutable
-        memory returned by the call may not be referenced by any other part of the
+        memory returned by the call cannot be referenced by any other part of the
         program, i.e. it is newly allocated by the function.
-        Nor may the mutable
-        references of the result refer to any object that
-        existed before the function call. For example:)
+        The mutable
+        references of the result similarly cannot refer to any object that
+        existed before the function call.
+        This allows the result to be implicitly cast to any qualifier.
+        For example:)
 
         $(SPEC_RUNNABLE_EXAMPLE_COMPILE
         ---
         struct List { int payload; List* next; }
+
         pure List* make(int a, int b)
         {
             auto result = new List(a, null);
             result.next = new List(b, result);
             return result;
         }
+
+        void main()
+        {
+            auto list = make(1, 2);
+            pragma(msg, typeof(list));       // List*
+
+            immutable ilist = make(1, 2);
+            pragma(msg, typeof(ilist));      // immutable List*
+            pragma(msg, typeof(ilist.next)); // immutable List*
+        }
         ---
         )
 
-        $(P All references in `make`'s result refer to other `List`
+        $(P All references in `make`'s result refer to `List`
         objects created by `make`, and no other part of the program refers to
-        any of these objects.
-        This restriction does not apply to any Exception or Error thrown from the function.
-        )
+        any of these objects. Hence the result can initialize an immutable
+        variable.)
 
+        $(P This does not affect any *Exception* or *Error* thrown from the function.
+        )
 
-        $(P Pure destructors do not benefit of special elision.)
+$(H3 $(LNAME2 pure-optimization, Optimization))
 
         $(IMPLEMENTATION_DEFINED An implementation may assume that a strongly pure
         function that returns a result
         without mutable indirections will have the same effect for all invocations
         with equivalent arguments. It is allowed to memoize the result of the
         function under the assumption that equivalent parameters always produce
-        equivalent results.
+        equivalent results.)
+
+        $(P
         A strongly pure function may still have behavior
         inconsistent with memoization by e.g. using `cast`s or by changing behavior
         depending on the address of its parameters. An implementation is currently
         not required to enforce validity of memoization in all cases.
-        If a strongly pure function throws an Exception or an Error, the
+        If a strongly pure function throws an *Exception* or an *Error*, the
         assumptions related to memoization do not carry to the thrown
         exception.)
 
+        $(P Pure destructors do not benefit of special elision.)
```

Comment:
I have no idea what this means, but it was already there so it's not up to this PR to clarify.